search-sentiments-map/public/scripts/map-script.js

```diff
@@ -76,37 +76,40 @@ function loadCountryDataByMode() {
   loadCountryData(isSentimentMode);
 }
 
-/** Loads the country sentiment score from Datastore. */
-function loadCountryData() {
-  var dataVariableMax;
-  var countryMax = '';
-  var dataVariableMin; 
-  var countryMin = '';
-  map.data.forEach(function (row) {
-    const countryCode = row.getId();
+/** 
+ * Loads the sentiments or search interests for all countries from Datastore 
+ * depending on what mode has been specified.
+ * @param {boolean=} isSentimentMode Whether the result to obtain is the sentiment
+ * data. Search interest data will be loaded otherwise.
+ */
+function loadCountryData(isSentimentMode = true) {
+
+  // Minimum and maximum average sentiment values for current topic. 
+  let dataVariableMax = Number.MIN_VALUE;
+  let dataVariableMin = Number.MAX_VALUE; 
+
+  // Countries with the minimum and maximum average sentiment values for current topic. 
+  let countryMax = '';
+  let countryMin = '';
+  map.data.forEach(function(row) {
+    let dataByCountry = getCurrentCustomSearchData().dataByCountry;
+    let countryData = dataByCountry.filter(data => data.country === row.getId());
     const country = row.getProperty('name');
-    let topicData = getCurrentCustomSearchData();
-    let countryData = topicData.countries
-      .filter(countries => countries.country === countryCode);
     let dataVariable = 0;
-    if (countryData.length != 0) {
-      let score = countryData[0].averageSentiment;
-      dataVariable = score;
-      if (dataVariableMax == undefined) {
-        dataVariableMax = score;
-        countryMax = country;
-        dataVariableMin = score;
-        countryMin = country;
-      } else if (score > dataVariableMax) {
-        dataVariableMax = score;
-        countryMax = country;
-      } else if (score < dataVariableMin) {
-        dataVariableMin = score;
-        countryMin = country;
-      }
+
+     // Handle case where there are no search results for the current topic.
+    if (countryData.length === 0) {
+      console.log('Data does not exist for this country:', country);
     } else {
       dataVariable = 
           isSentimentMode ? countryData[0].averageSentiment : countryData[0].interest;
+      if (dataVariable > dataVariableMax) {
+        dataVariableMax = dataVariable;
+        countryMax = country;
+      } else if (dataVariable < dataVariableMin) {
```

Comment:
What happens if the user is searching a custom topic, and they are only viewing results for one country. What will happen to the max and min here?

Reply:
The max and min will be that one country, as expected

Reply:
Okay sounds good.

Reply:
Actually, since you use `else if` here, won't it only set the country to be the max, and not the min?